# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


In [1]:
import numpy as np
from scipy import stats
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

In [5]:
# Дані з умови
time_to_purchase = np.array([128, 135, 121, 142, 126, 151, 123])
mu0 = 123.7
alpha = 0.10

Гіпотези

H₀: μ = 123.7 (середній час від першого заходу на сайт до покупки дорівнює 123.7 хвилин)

H₁: μ > 123.7 (середній час вище за 123.7)

Тип тесту: правосторонній (one-tailed)

In [9]:
# Обчислимо вибіркову статистику
x_bar = np.mean(time_to_purchase)
s = np.std(time_to_purchase, ddof=1)
n = len(time_to_purchase)

print(f"Вибіркове середнє: {x_bar:.2f}")
print(f"Вибіркове стандартне відхилення: {s:.2f}")
print(f"Величина вибірки: {n}")

Вибіркове середнє: 132.29
Вибіркове стандартне відхилення: 10.98
Величина вибірки: 7


In [7]:
# Критичне значення
df = n - 1
t_crit = stats.t.ppf(1 - alpha, df)
print(f"Критичне значення (df={df}): {t_crit:.3f}")

Критичне значення (df=6): 1.440


In [10]:
# scipy тест
res = stats.ttest_1samp(time_to_purchase, popmean=mu0, alternative="greater")
print(f"scipy: t = {res.statistic:.3f}, p-value = {res.pvalue:.4f}")

scipy: t = 2.069, p-value = 0.0420


p-value: 0.0420

Порівнюємо: 0.0420 < 0.10

p-value менше за рівень значущості

Відхиляємо H0: середній time-to-purchase менший за 123.7 на рівні значущості 0.10

На рівні значущості:

α=0.10

є статистично значущі докази того, що після редизайну середній час до покупки збільшився.

**Чи зміниться ваше рішення при зміні рівня значущості на  𝛼=0.05?**

Ні, не зміниться, оскільки 0.0420 все ще менше за 0.05

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [12]:
count = 311
nobs = 1200
p0 = 0.23

# z-тест для пропорції
stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')
confint = proportion_confint(count, nobs, alpha=0.03, method='normal')

print(f"Z-статистика = {stat:.3f}")
print(f"p-value = {pval:.4f}")
print(f"Довірчий інтервал: {confint}")

Z-статистика = 2.306
p-value = 0.0106
Довірчий інтервал: (0.23171700302179205, 0.28661633031154127)


Це односторонній тест

𝐻0:𝑝=0.23

𝐻1:𝑝>0.23

Оскільки  𝑝-value=0.0106 < 0.03 , відхиляємо  𝐻0 . Маємо статистично значущі докази, що впізнаваність бренду зросла і стала більша за 23% після рекламної кампанії.

Довірчий інтервал (0.232, 0.287), це означає, що з ймовірністю 95% частка лежить між 23,2% і 28,7%. Початкова границя інтервалу вже незначно вища за 23%, це свідчить на користь висновків вище, що бренд стали впізнавати більша частка дорослих людей.